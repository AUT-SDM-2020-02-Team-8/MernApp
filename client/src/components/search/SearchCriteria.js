import React, { Component } from "react"
import { Grid } from "@material-ui/core"
import SearchRow from "./SearchRow"

class SearchCriteria extends Component {
  constructor() {
    super()
    this.state = {
      criteria: [],
      counter: 0
    }
    this.addSearchRow = this.addSearchRow.bind(this)
    this.removeSearchRow = this.removeSearchRow.bind(this)
  }

  componentDidMount(){
    this.addSearchRow()
  }

  addSearchRow() {
    this.setState({
      criteria: [...this.state.criteria, <SearchRow key={this.state.counter} counter={this.state.counter} addSearchRow={this.addSearchRow} removeSearchRow={this.removeSearchRow} />],
      counter: this.state.counter + 1
    })
  }

  removeSearchRow(counter) {
    const newCriteria = this.state.criteria.filter(criteria => criteria.key !== counter.toString())
    if (newCriteria.length > 0) {
      this.setState({
        criteria: newCriteria
      })
    }
  }
  
  render() {
    return (
      <Grid item xs={12}>
        {this.state.criteria}
      </Grid>
    )
  }
}

export default SearchCriteria