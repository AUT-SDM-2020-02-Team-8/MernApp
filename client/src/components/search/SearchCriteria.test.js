import React from 'react';
import { shallow } from 'enzyme';
import SearchCriteria from './SearchCriteria';

test('renders homepage', () => {
  const wrapper = shallow(<SearchCriteria />);
  expect(wrapper).toMatchSnapshot();
});
