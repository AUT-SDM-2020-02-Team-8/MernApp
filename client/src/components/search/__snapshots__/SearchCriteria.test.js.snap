// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  xs={12}
>
  <SearchRow
    addSearchRow={[Function]}
    counter={0}
    key="0"
    removeSearchRow={[Function]}
  />
</WithStyles(ForwardRef(Grid))>
`;
